backend/controllers/groups-controllers.js

```diff
@@ -0,0 +1,199 @@
+const express = require("express");
+const Group = require("../database/models/groups");
+const GroupClaims = require("../database/models/group_claims");
+const AdvancedSettings = require("../database/models/advanced_settings");
+const { v4: uuidv4 } = require("uuid");
+const { Op } = require("sequelize");
+const models = require("../database/models");
+
+const createGroup = async (req, res) => {
+  const group_id = uuidv4();
+  const groupName = req.body.name;
+  const creationDate = req.body.creationDate;
+  const createdBy = req.body.createdBy;
+  const claims = req.body.claims;
+  const key = req.body.advancedSettings.key;
+  const value = req.body.advancedSettings.value;
+
+  await models.Group.create({
+    group_id: group_id,
+    name: groupName,
+    creationDate: creationDate,
+    createdBy: createdBy,
+  }).catch((err) => {
+    console.log("Error: " + err);
+    res.sendStatus(400);
+  });
+
+  await models.GroupClaims.create({
+    group_id: group_id,
+    claims: claims,
+  }).catch((err) => {
+    console.log("Error: " + err);
+    res.sendStatus(400);
+  });
+
+  await models.AdvancedSettings.create({
+    group_id: group_id,
+    key: key,
+    value: value,
+  })
+    .then(res.sendStatus(200))
+    .catch((err) => {
+      console.log("Error: " + err);
+      res.sendStatus(400);
+    });
+};
+
+const deleteGroup = async (req, res) => {
+  const groupId = req.params.groupId;
+
+  await models.Group.destroy({
+    where: {
+      group_id: {
+        [Op.eq]: groupId,
+      },
+    },
+  }).catch((err) => {
+    console.log("Error: " + err);
+    res.sendStatus(400);
+  });
+
+  await models.GroupClaims.destroy({
+    where: {
+      group_id: {
+        [Op.eq]: groupId,
+      },
+    },
+  }).catch((err) => {
+    console.log("Error: " + err);
+    res.sendStatus(400);
+  });
+
+  await models.AdvancedSettings.destroy({
+    where: {
+      group_id: {
+        [Op.eq]: groupId,
+      },
+    },
+  })
+    .then(res.sendStatus(200))
+    .catch((err) => {
+      console.log("Error: " + err);
+      res.sendStatus(400);
+    });
+};
+
+const getGroups = async (req, res) => {
+  let groups;
+  try {
+    groups = await models.Group.findAll({
+      include: [
+        {
+          model: models.GroupClaims,
+          as: "claims",
+        },
+        {
+          model: models.AdvancedSettings,
+          as: "advancedSettings",
+        },
+      ],
+    });
+
+    return res.status(200).json({ groups });
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+const getGroupsById = async (req, res) => {
+  const { groupId } = req.params;
```

Comment:
What if params does not exist? or is invalid? or is valid but is not stored in database?